server/routes.js

```diff
@@ -0,0 +1,254 @@
+				.then(fillAuthorInfo(db))
+				.then((posts) => {
+					posts.forEach((post) => post.short = true);
+					return db.posts.count()
+						.then((count) => ({
+							title: "",
+							posts,
+							pagination: {
+								left: pageNumber > 1 ? `/archive/${pageNumber - 1}` : undefined,
+								right: count > pageNumber * 5 ? `/archive/${pageNumber + 1}` : undefined
+							}
+						}))
+				})
+				.then(fillDefaultSidebar(db))
+		}
+	},
+	{
+		path:/^\/editor(\/([0-9a-zA-Z_-]{7,14}))?$/,
+		page: "editor.hbs",
+		cache: false,
+		context: ([_, __, postId], db) =>
+			db.posts.findOne({ guid: postId })
+				.then((post) => {
+					if (post !== undefined) {
+						return {
+							title: `Editing "${post.title}"`,
+							post,
+							admin: true
+						}
+					} else {
+						return {
+							title: `Editing New Post`,
+							admin: true
+						}
+					}
+				})
+	},
+	{
+		path:/^\/tag\/([a-z0-9\-]{1,16})$/,
+		page: "page.hbs",
+		cache: true,
+		prerender: (db) => {
+			return db.posts.distinct("tags", {})
+				.then((tags) => tags.map((tag) => `/tag/${tag}`))
+		},
+		context: ([_, tag], db) =>
+			aggregatePosts(db, { $match: { tags: tag } }, { $sort: { timestamp: -1 } })
+				.then(fillAuthorInfo(db))
+				.then((posts) => {
+					posts.forEach((post) => post.short = true);
+					return {
+						title: `Tagged "${tag}"`,
+						posts: posts
+					};
+				})
+				.then(fillDefaultSidebar(db))
+	},
+	{
+		path:/^\/author\/(matt|zach|ben|danial|david|ellis|kyle|thomas)$/, // sigh...
+		page: "page.hbs",
+		cache: true,
+		prerender: Object.keys(config.adminInfo).map((author) => `/author/${author}`),
+		context: ([_, author], db) =>
+			aggregatePosts(db, { $match: { author: users[author].gid } }, { $sort: { timestamp: -1 } })
+				.then(fillAuthorInfo(db))
+				.then((posts) => {
+					posts.forEach((post) => post.short = true);
+					return {
+						title: `Posts by ${author}`,
+						posts: posts,
+						filter: {
+							type: "author",
+							author: users[author]
+						}
+					};
+				})
+				.then(fillDefaultSidebar(db))
+	},
+	{
+		path:/^\/posts\/([0-9a-zA-Z_-]{7,14})$/,
+		page: "page.hbs",
+		cache: true,
+		prerender: (db) => {
+			return db.posts.find({})
+				.then((posts) => posts.map((post) => `/posts/${post.guid}`));
+		},
+		context: ([_, postId], db) =>
+			db.posts.findOne({ guid: postId })
+				.then(fillAuthorInfo(db))
+				.then((post) => ({
+						title: post.title.text,
+						posts: [post]
+					}))
+	},
+	{
+		path:/^\/preview\/([0-9a-zA-Z_-]{7,14})$/,
+		page: "page.hbs",
+		cache: false,
+		context: ([_, postId], db) =>
+			db.posts.findOne({ guid: postId })
+				.then(fillAuthorInfo(db))
+				.then((post) => {
+					post.hideComments = true;
+					return {
+						title: post.title.text,
+						posts: [post]
+					};
+				})
+	},
+	{
+		path:/^\/raw\/([0-9a-zA-Z_-]{7,14})$/,
+		page: "raw.hbs",
+		cache: true,
+		prerenderFIXME: (db) => {
```

Comment:
That was my way of "commenting out" anything that was still really broken in the renderer so that it wouldn't get prevent me from testing other pages.  Will fix.